enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,188 @@
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Devlopers will be able to remove any of the currently managed CVO objects by modifying an existing manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for managment by CVO, therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+This enhancement neither adds nor requires any application specific delete logic within CVO.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO.
+Values other than `true` will result in a CVO error and no further processing of the manifest.
+```
+apiVersion: apps/v1
+...
+metadata:
+...
+  annotations:
+    release.openshift.io/delete: "true"
+```
+The existing CVO ordering scheme defined [here](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/operators.md#what-is-the-order-that-resources-get-createdupdated-in) will also be used for object removal.
+This both minimizes CVO code changes and provides a simple method of deleting multiple objects by reversing the order in which they were created.
+It is the developer's responsibility to ensure proper deletion ordering and to ensure that all items originally created by an object are deleted when that object is deleted.
+For example, an operator may have to be modified to take explicit actions upon delete request to remove external resources it had originally created.
+
+### Delete Finalization
+
+Two delete finalization options are decribed below to help reach a consensus on the best approach.
+
+#### Finalization Option 1
+
+The first approach is to handle deletion requests similar to how CVO handles create/update requests.
+This is in a non-blocking manner whereby CVO issues the initial request to delete an object kicking off resource finalization and afterwhich resource removal.
+CVO does not wait for actual resource removal but instead continues.
+As CVO manifest processing continues the object's `.metadata.deletionTimestamp`, set when the delete was originally requested, informs CVO that this object has already been processed for deletion.
+If an object cannot be successfully removed cluster update will continue since the object's presence, or partial removal, should not affect cluster operation.
+
+#### Finalization Option 2
+
+The second approach is to handle deletion requests synchronously whereby CVO will wait for confirmation that the given object has been removed before continuing through the manifest graph.
+Again, since the object is being removed neither its presence, nor lack thereof, should affect cluster operation.
+Therefore CVO will wait a limited time for object removal and then continue with the manifest graph and the update.
```

Comment:
What is advantage of option 2 on option1 or vice versa?

Reply:
Adding advantages in commit [37a9487](https://github.com/openshift/enhancements/pull/419/commits/37a9487976a935ca108c703f38868d8743177b47).